include/dkm.hpp

```diff
@@ -87,6 +87,13 @@ std::vector<std::array<T, N>> random_plusplus(const std::vector<std::array<T, N>
 		// Calculate the distance to the closest mean for each data point
 		auto distances = details::closest_distance(means, data);
 		// Pick a random point weighted by the distance from existing means
+		// If all distances are 0, the normalization step in std::discrete_distribution can cause a floating point
+		// exception, thus we check that:
+		double distance_sum = std::accumulate(distances.begin(), distances.end(), 0.0);
```

Comment:
This doesn't check that all that the distances are 0, it just checks that they _sum_ to 0. A `std::all_of` with a predicate would check _all_ values are zero.